conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt

```diff
@@ -309,57 +286,40 @@ class ExternalDatabasePermissioner(
                 hikariDataSource = extDbManager.connectToOrg(collaborationId),
-                tableSchema = schema,
-                tableName = table.name,
-                columns = columns,
+                columnAclKeyToName = columnAclKeyToName,
                 permissions = allViewPermissions
-
         )
     }
 
     private fun initializePermissionSetForExternalTable(
             hikariDataSource: HikariDataSource,
-            tableSchema: String,
-            tableName: String,
-            columns: Set<ExternalColumn>,
+            columnAclKeyToName: Map<AclKey, String>,
             permissions: Set<Permission>
     ) {
-        val targetsToColumns = mutableMapOf<AccessTarget, ExternalColumn>()
-
-        val targets = columns.flatMapTo(mutableSetOf()) { column ->
-            permissions.map { permission ->
-                val at = AccessTarget.forPermissionOnTarget(permission, column.tableId, column.id)
-                targetsToColumns[at] = column
-                at
-            }
-        }
-
-        PostgresRoles.getOrCreatePermissionRolesAsync(
-                externalRoleNames,
-                targets,
-                hikariDataSource
-        ).thenApplyAsync { targetToRoleNames ->
-            targetToRoleNames.map { (target, roleName) ->
-                val column = targetsToColumns.getValue(target)
-                val pgPermissions = olToPostgres.getValue(target.permission)
-                grantPermissionsOnColumnsOnTableToRoleSql(
-                        pgPermissions,
-                        ApiHelpers.dbQuote(column.name),
-                        tableSchema,
-                        tableName,
-                        roleName.toString()
-                )
-            }
-        }.thenAccept { sqls ->
-            hikariDataSource.connection.use { conn ->
-                conn.createStatement().use { stmt ->
-                    sqls.forEach { sql ->
-                        stmt.addBatch(sql)
+        // insert column name into external_permission_roles as role_id, needs more thinking
+        // actually, why do we even need external_permission_roles? isn't information_schema.column_privileges enough?
+        hikariDataSource.connection.use { conn ->
+            conn.createStatement().use { stmt ->
+                permissions.forEach { permission ->
+                    columnAclKeyToName.filterNot {
+                        externalRoleNames.containsKey(AccessTarget(it.key, permission))
+                    }.forEach { 
+                        olToPostgres.getValue(permission).forEach { privilege -> 
+                            val newRoleId = UUID.randomUUID()
+                            stmt.addBatch("""
+                                INSERT INTO ${HazelcastMap.EXTERNAL_PERMISSION_ROLES.name}
+                                VALUES ('{\"${it.key}\"}'::uuid[], \"$privilege\", ${ApiHelpers.dbQuote(it.value)}, \"${newRoleId}\"::uuid) ON CONFLICT DO NOTHING
```

Comment:
1. It turns out double quote is not required, huh I didn't know that!
2. I made a mistake here, thanks for catching it! It should be:
```suggestion
                                VALUES ('{\"${it.key.joinToString("\",\"")}\"}'::uuid[], \"$privilege\", ${ApiHelpers.dbQuote(it.value)}, \"${newRoleId}\"::uuid) ON CONFLICT DO NOTHING
```
Cleaner way is to add a `toPostgres` function to `AclKey` class.

Reply:
Ok, for now, all casting of `AclKey` for Postgresql is done through: https://github.com/openlattice/openlattice/blob/bf948e34a0d2ce07a3346f2303f512f086508258/conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt#L228-L230

Still think maybe adding it to `AclKey` is more correct, but this will do for now. What do you think?

Reply:
This is fine for now.